llvm/lib/ObjectYAML/DXContainerYAML.cpp

```diff
@@ -31,11 +34,34 @@ DXContainerYAML::ShaderFeatureFlags::ShaderFeatureFlags(uint64_t FlagData) {
 
 DXContainerYAML::RootSignatureYamlDesc::RootSignatureYamlDesc(
     const object::DirectX::RootSignature &Data)
-    : Version(Data.getVersion()), NumParameters(Data.getNumParameters()),
-      RootParametersOffset(Data.getRootParametersOffset()),
+    : Version(Data.getVersion()),
       NumStaticSamplers(Data.getNumStaticSamplers()),
       StaticSamplersOffset(Data.getStaticSamplersOffset()) {
   uint32_t Flags = Data.getFlags();
+  for (const auto &PH : Data.param_header()) {
+
+    auto NewP = RootParameterYamlDesc();
```

Comment:
I think that this is just constructing a new `RootParameterYamlDesc` on the stack?

```suggestion
    RootParameterYamlDesc NewP;
```